daily-rotate-file.js

```diff
@@ -20,134 +18,167 @@ var loggerDefaults = {
             watch_log: options.watchLog ? options.watchLog : false,
-            audit_hash_type: options.auditHashType ? options.auditHashType  : 'sha256'
+            audit_hash_type: options.auditHashType ? options.auditHashType : "sha256"
         });
 
-        this.logStream.on('new', function (newFile) {
-            self.emit('new', newFile);
+        this.logStream.on("new", (newFile) => {
+            this.emit("new", newFile);
         });
 
-        this.logStream.on('rotate', function (oldFile, newFile) {
-            self.emit('rotate', oldFile, newFile);
+        this.logStream.on("rotate", (oldFile, newFile) => {
+            this.emit("rotate", oldFile, newFile);
         });
 
-        this.logStream.on('logRemoved', function (params) {
+        this.logStream.on("logRemoved", (params) => {
             if (options.zippedArchive) {
-                var gzName = params.name + '.gz';
-                if (fs.existsSync(gzName)) {
-                    try {
-                        fs.unlinkSync(gzName);
-                    }
-                    catch (_err) {
-                        // file is there but we got an error when trying to delete,
-                        // so permissions problem or concurrency issue and another
-                        // process already deleted it we could detect the concurrency
-                        // issue by checking err.type === ENOENT or EACCESS for
-                        // permissions ... but then?
+                const gzName = params.name + ".gz";
+                try {
+                    fs.unlinkSync(gzName);
+                } catch (err) {
+                    // ENOENT is okay, means file doesn't exist, other errors prevent deletion, so report it
+                    if (err.code !== "ENOENT") {
+                        err.message = `Error occurred while removing ${gzName}: ${err.message}`;
+                        this.emit("error", err);
+                        return;
                     }
-                    self.emit('logRemoved', gzName);
-                    return;
                 }
+                this.emit("logRemoved", gzName);
+                return;
             }
-            self.emit('logRemoved', params.name);
+            this.emit("logRemoved", params.name);
         });
 
         if (options.zippedArchive) {
-            this.logStream.on('rotate', function (oldFile) {
-                var oldFileExist = fs.existsSync(oldFile);
-                var gzExist = fs.existsSync(oldFile + '.gz');
-                if (!oldFileExist || gzExist) {
+            this.logStream.on("rotate", (oldFile) => {
+                try {
+                    if (!fs.existsSync(oldFile)) {
+                        return;
+                    }
+                } catch (err) {
+                    err.message = `Error occurred while checking existence of ${oldFile}: ${err.message}`;
+                    this.emit("error", err);
```

Comment:
In the README you state, make sure to add a listener, else your application crashes. But also if there is no listener the error is not re-thrown. Or did I misunderstand you here? 

Else we could check for listeners, if there emit else throw.

Reply:
Node.js does that automatically ... if there is an error event which is not listened too it will be thrown. see https://nodejs.org/api/events.html#error-events ... so it is an implicit throw